ledger/test-common/src/main/daml/model/Foo.daml

```diff
@@ -3,6 +3,50 @@
 
 module Foo where
 
+import DA.Functor (void)
+
+
+template Divulger
+  with
+    divulgees: [Party] -- Parties to whom something is divulged
```

Comment:
The divulgees are used in the signatory clause which I assume will only allow the contract to be created if all the divulgees are in the `actAs` set of parties.  

I see the Foo creations are divulging the created contract to the observers.

I don't see any interplay between the divulgees and the created contracts, am I missing something ?

Reply:
> The divulgees are used in the signatory clause which I assume will only allow the contract to be created if all the divulgees are in the actAs set of parties.
Yes. Note that `divulgees` here can be any subset of all allocated divulgee parties.
When submitting a command I'm always adding all divulgee parties to the `actAs` set of submitters so I can authorize creation of any contract.

> I see the Foo creations are divulging the created contract to the observers.
Yes, Foo creations are visible to observers, but I wouldn't call it divulgences. It's just how contract observer's behave.

> I don't see any interplay between the divulgees and the created contracts, am I missing something ?
`divulgees` are in the signatories on a `Divulger` contract so they see the exercises of the nonconsuming `DivulgeImmediate`. 
`divulgees` can see creations of `Foo1/Foo2/Foo3` contracts solely because these creations are a sub-action of `DivulgeImmediate`.